utils/s2n_array.c

```diff
@@ -144,3 +144,26 @@ int s2n_array_free(struct s2n_array *array)
 
     return 0;
 }
+
+int s2n_array_binary_search(int low, int top, struct s2n_array *array, void *element,
```

Comment:
Add some tests for it in `s2n_array_test.c`?

Reply:
Oops. Forgot about that.
Done.